In [1]:
import numpy as np 
import pandas as pd
import matplotlib as plt

import csv
import matplotlib.pyplot as plt
from sklearn import (
    linear_model, metrics, pipeline, model_selection
)
from sklearn.linear_model import LogisticRegression

import statsmodels.api as sm
import statsmodels.formula.api as smf

import linearmodels

import seaborn as sns

import statsmodels.api as sm
import statsmodels.formula.api as smf

import matplotlib.patches as mpatches

In [2]:
import os
os.getcwd()

'C:\\Users\\jonah\\Documents\\Schoolwork\\ECON 594\\ECON_594_WD'

In [3]:
#import os
#os.getcwd()
#path="C:\\Users\\jonah\\Documents\\Schoolwork\\ECON 531\\data\\Working Folder"
#path="C:\\Users\\jonah\\Documents\\Schoolwork\\CPSC 330\\cpsc330-2021W2"
#os.chdir(path)
#Documents/Schoolwork/ECON 594/ECON_594_WD/LTDB/LTDB_Std_1970_Sample.csv

# Importing and Filtering 1970 Data

In [28]:
ltdb_1970 = pd.read_csv("LTDB\LTDB_Std_1970_Sample.csv")

C:\Users\jonah\anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3012: DtypeWarning: Columns (8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,49,51,60) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [29]:
ltdb_1970.columns

Index(['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', 'MEX70', 'PR70', 'CUBAN70', 'FAMILY70', 'FHH70',
       'POP70SP2', 'RUANC70', 'ITANC70', 'GEANC70', 'IRANC70', 'SCANC70',
       'RUFB70', 'ITFB70', 'GEFB70', 'IRFB70', 'SCFB70', 'FB70', 'NAT70',
       'POP70SP1', 'N10IMM70', 'AG25UP70', 'HS70', 'COL70', 'CLF70', 'UNEMP70',
       'DFLABF70', 'FLABF70', 'EMPCLF70', 'PROF70', 'MANUF70', 'SEMP70',
       'AG16CV70', 'VET70', 'CNI16U70', 'DIS70', 'DPOV70', 'NPOV70',
       'N65POV70', 'DFMPOV70', 'NFMPOV70', 'NWPOV70', 'DWPOV70', 'NBPOV70',
       'DBPOV70', 'INCPC70', 'HU70SP', 'H30OLD70', 'OHU70SP', 'H10YRS70',
       'HINC70', 'MHMVAL70', 'MRENT70', 'HH70'],
      dtype='object')

In [30]:
print(len(ltdb_1970["TRTID10"].unique()))
print(len(ltdb_1970["tract"].unique()))
print(len(ltdb_1970["placefp10"].unique()))

52759
22073
5696


In [6]:
ltdb_1970 = ltdb_1970[['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', "MHMVAL70"]]

In [7]:
ltdb_1970

TRTID10 state            county               tract  placefp10  \
0       1001020500    AL    Autauga County    Census Tract 205      62328   
1       1003010100    AL    Baldwin County    Census Tract 101      99999   
2       1003010200    AL    Baldwin County    Census Tract 102      99999   
3       1003010300    AL    Baldwin County    Census Tract 103      99999   
4       1003010400    AL    Baldwin County    Census Tract 104      99999   
...            ...   ...               ...                 ...        ...   
52754  55139003400    WI  Winnebago County     Census Tract 34      55750   
52755  55139003500    WI  Winnebago County     Census Tract 35      55750   
52756  55139003600    WI  Winnebago County     Census Tract 36      55750   
52757  55139003701    WI  Winnebago County  Census Tract 37.01      55750   
52758  55139003702    WI  Winnebago County  Census Tract 37.02      55750   

       cbsa10  metdiv10  ccflag10     MHMVAL70  
0       33860     99999         0  14141.79117  
1       19300     99999         0  5441.176129  
2       19300     99999         0  6961.257952  
3       19300     99999         0  8675.884928  
4       19300     99999         0  7596.191353  
...       ...       ...       ...          ...  
52754   36780     99999         1  15646.88635  
52755   36780     99999         1  19160.15698  
52756   36780     99999         1   27924.7745  
52757   36780     99999         1  21189.65493  
52758   36780     99999         1   21035.3701  

[52759 rows x 9 columns]

# Importing and Filtering 1980 Data

In [8]:
ltdb_1980 = pd.read_csv("LTDB\LTDB_Std_1980_fullcount.csv")

In [9]:
ltdb_1980.columns

Index(['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', 'POP80', 'NHWHT80', 'NHBLK80', 'NTV80',
       'ASIAN80', 'HISP80', 'HAW80', 'INDIA80', 'CHINA80', 'FILIP80',
       'JAPAN80', 'KOREA80', 'VIET80', 'MEX80', 'PR80', 'CUBAN80', 'HU80',
       'VAC80', 'OHU80', 'A18UND80', 'A60UP80', 'A75UP80', 'AG15UP80', 'MAR80',
       'WDS80', 'MHMVAL80', 'MRENT80', 'OWN80', 'RENT80', 'GlobD80',
       'GlobG80'],
      dtype='object')

In [10]:
ltdb_1980 = ltdb_1980[['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', "MHMVAL80"]]

In [11]:
ltdb_1980

TRTID10 state                county                tract  \
0       1001020100    AL        Autauga County     Census Tract 201   
1       6083002402    CA  Santa Barbara County   Census Tract 24.02   
2       1001020200    AL        Autauga County     Census Tract 202   
3       6083002102    CA  Santa Barbara County   Census Tract 21.02   
4       1001020300    AL        Autauga County     Census Tract 203   
...            ...   ...                   ...                  ...   
59182  26081011804    MI           Kent County  Census Tract 118.04   
59183  26077002901    MI      Kalamazoo County   Census Tract 29.01   
59184  26077002903    MI      Kalamazoo County   Census Tract 29.03   
59185  47149040810    TN     Rutherford County  Census Tract 408.10   
59186  53073010100    WA        Whatcom County     Census Tract 101   

       placefp10  cbsa10  metdiv10  ccflag10     MHMVAL80  
0          62328   33860     99999         0          NaN  
1          69196   42060     99999         1  65120.99963  
2          62328   33860     99999         0  26837.00034  
3          69196   42060     99999         1  66500.00045  
4          62328   33860     99999         0          NaN  
...          ...     ...       ...       ...          ...  
59182      99999   24340     99999         0  58685.99943  
59183      99999   28020     99999         0  45606.85763  
59184      99999   28020     99999         0  58217.46546  
59185      99999   34980     99999         0  40642.00092  
59186      53800   13380     99999         0  46142.94263  

[59187 rows x 9 columns]

# Importing and Filtering 1990 Data

In [31]:
ltdb_1990 = pd.read_csv("LTDB\LTDB_Std_1990_fullcount.csv")

In [33]:
ltdb_1990.columns

Index(['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', 'POP90', 'NHWHT90', 'NHBLK90', 'NTV90',
       'ASIAN90', 'HISP90', 'HAW90', 'INDIA90', 'CHINA90', 'FILIP90',
       'JAPAN90', 'KOREA90', 'VIET90', 'MEX90', 'PR90', 'CUBAN90', 'HU90',
       'VAC90', 'OHU90', 'A18UND90', 'A60UP90', 'A75UP90', 'AGEWHT90',
       'A15WHT90', 'A60WHT90', 'AGEBLK90', 'A15BLK90', 'A60BLK90', 'AGEHSP90',
       'A15HSP90', 'A60HSP90', 'AGENTV90', 'A15NTV90', 'A60NTV90', 'AGEASN90',
       'A15ASN90', 'A60ASN90', 'AG15UP90', 'MAR90', 'WDS90', 'MHMVAL90',
       'MRENT90', 'OWN90', 'RENT90', 'GlobD90', 'GlobG90'],
      dtype='object')

In [34]:
ltdb_1990 = ltdb_1990[['TRTID10', 'state', 'county', 'tract', 'placefp10', 'cbsa10',
       'metdiv10', 'ccflag10', "MHMVAL90"]]

In [36]:
ltdb_1990

TRTID10 state                county               tract  placefp10  \
0       1001020100    AL        Autauga County    Census Tract 201      62328   
1       6083002402    CA  Santa Barbara County  Census Tract 24.02      69196   
2       1001020200    AL        Autauga County    Census Tract 202      62328   
3       6083002102    CA  Santa Barbara County  Census Tract 21.02      69196   
4       1001020300    AL        Autauga County    Census Tract 203      62328   
...            ...   ...                   ...                 ...        ...   
72688  53065950100    WA        Stevens County   Census Tract 9501      99999   
72689  53047970500    WA       Okanogan County   Census Tract 9705      99999   
72690  53057951200    WA         Skagit County   Census Tract 9512       6050   
72691  53057951000    WA         Skagit County   Census Tract 9510      99999   
72692  53073010100    WA        Whatcom County    Census Tract 101      53800   

       cbsa10  metdiv10  ccflag10      MHMVAL90  
0       33860     99999         0   62200.00055  
1       42060     99999         1  144999.99920  
2       33860     99999         0   43600.00056  
3       42060     99999         1  157600.00070  
4       33860     99999         0   55200.00044  
...       ...       ...       ...           ...  
72688   99999     99999         0   46512.35679  
72689   99999     99999         0   55472.13408  
72690   34580     99999         0  100351.86530  
72691   34580     99999         0   58799.99740  
72692   13380     99999         0   71300.08319  

[72693 rows x 9 columns]

# Comparing counts of `TRTID10` and `tract`

In [12]:
ltdb_1970["TRTID10"].nunique()

52759

In [13]:
ltdb_1980["TRTID10"].nunique()

59187

In [37]:
ltdb_1990["TRTID10"].nunique()

72693

In [14]:
ltdb_1970["tract"].nunique()

22073

In [15]:
ltdb_1980["tract"].nunique()

22786

In [38]:
ltdb_1990["tract"].nunique()

23795

# Joining on TRTID10

In [16]:
ltdb_1980["TRTID10"].dtypes

dtype('int64')

In [17]:
ltdb_1970["TRTID10"].dtypes

dtype('int64')

In [18]:
ltdb_1970_1980 = ltdb_1980.merge(ltdb_1970, on="TRTID10")
ltdb_1970_1980

TRTID10 state_x              county_x              tract_x  \
0       6083002402      CA  Santa Barbara County   Census Tract 24.02   
1       6083002102      CA  Santa Barbara County   Census Tract 21.02   
2       6083002009      CA  Santa Barbara County   Census Tract 20.09   
3       6083002008      CA  Santa Barbara County   Census Tract 20.08   
4       1001020500      AL        Autauga County     Census Tract 205   
...            ...     ...                   ...                  ...   
52754  26081011801      MI           Kent County  Census Tract 118.01   
52755  26081012605      MI           Kent County  Census Tract 126.05   
52756  26081011804      MI           Kent County  Census Tract 118.04   
52757  26077002901      MI      Kalamazoo County   Census Tract 29.01   
52758  26077002903      MI      Kalamazoo County   Census Tract 29.03   

       placefp10_x  cbsa10_x  metdiv10_x  ccflag10_x     MHMVAL80 state_y  \
0            69196     42060       99999           1  65120.99963      CA   
1            69196     42060       99999           1  66500.00045      CA   
2            54120     42060       99999           0  78422.00169      CA   
3            54120     42060       99999           0          NaN      CA   
4            62328     33860       99999           0  46618.08749      AL   
...            ...       ...         ...         ...          ...     ...   
52754        99999     24340       99999           0  58239.17621      MI   
52755        42820     24340       99999           0  54064.14933      MI   
52756        99999     24340       99999           0  58685.99943      MI   
52757        99999     28020       99999           0  45606.85763      MI   
52758        99999     28020       99999           0  58217.46546      MI   

                   county_y              tract_y  placefp10_y  cbsa10_y  \
0      Santa Barbara County   Census Tract 24.02        69196     42060   
1      Santa Barbara County   Census Tract 21.02        69196     42060   
2      Santa Barbara County   Census Tract 20.09        54120     42060   
3      Santa Barbara County   Census Tract 20.08        54120     42060   
4            Autauga County     Census Tract 205        62328     33860   
...                     ...                  ...          ...       ...   
52754           Kent County  Census Tract 118.01        99999     24340   
52755           Kent County  Census Tract 126.05        42820     24340   
52756           Kent County  Census Tract 118.04        99999     24340   
52757      Kalamazoo County   Census Tract 29.01        99999     28020   
52758      Kalamazoo County   Census Tract 29.03        99999     28020   

       metdiv10_y  ccflag10_y     MHMVAL70  
0           99999           1  20788.04248  
1           99999           1  19300.31598  
2           99999           0  21829.09451  
3           99999           0  20737.99024  
4           99999           0  14141.79117  
...           ...         ...          ...  
52754       99999           0  23154.94228  
52755       99999           0  22263.08727  
52756       99999           0  23361.20365  
52757       99999           0  19797.79479  
52758       99999           0  23144.38515  

[52759 rows x 17 columns]

In [19]:
ltdb_1970_1980["tract_x"] == ltdb_1970_1980["tract_y"]

0        True
1        True
2        True
3        True
4        True
         ... 
52754    True
52755    True
52756    True
52757    True
52758    True
Length: 52759, dtype: bool

In [20]:
ltdb_1970_1980["tract_x"].equals(ltdb_1970_1980["tract_y"])

True

This is just a check that the tracts are all the same.

In [21]:
#unnecessary
ltdb_1970_1980_tract = ltdb_1980.merge(ltdb_1970, on="tract")
ltdb_1970_1980_tract

TRTID10_x state_x        county_x                 tract  \
0         1001020100      AL  Autauga County      Census Tract 201   
1         1001020100      AL  Autauga County      Census Tract 201   
2         1001020100      AL  Autauga County      Census Tract 201   
3         1001020100      AL  Autauga County      Census Tract 201   
4         1001020100      AL  Autauga County      Census Tract 201   
...              ...     ...             ...                   ...   
1139454  18097360301      IN   Marion County  Census Tract 3603.01   
1139455  18081610802      IN  Johnson County  Census Tract 6108.02   
1139456  18097390403      IN   Marion County  Census Tract 3904.03   
1139457  18097330108      IN   Marion County  Census Tract 3301.08   
1139458  18097361400      IN   Marion County     Census Tract 3614   

         placefp10_x  cbsa10_x  metdiv10_x  ccflag10_x     MHMVAL80  \
0              62328     33860       99999           0          NaN   
1              62328     33860       99999           0          NaN   
2              62328     33860       99999           0          NaN   
3              62328     33860       99999           0          NaN   
4              62328     33860       99999           0          NaN   
...              ...       ...         ...         ...          ...   
1139454        36003     26900       99999           1  33551.99972   
1139455        25450     26900       99999           0  53428.99863   
1139456        36003     26900       99999           1  59491.99762   
1139457        36003     26900       99999           1          NaN   
1139458        36003     26900       99999           1          NaN   

           TRTID10_y state_y              county_y  placefp10_y  cbsa10_y  \
0         1033020100      AL        Colbert County        69648     22520   
1         1127020100      AL         Walker County        38416     13820   
2         5033020100      AR       Crawford County        99999     22900   
3         5091020100      AR         Miller County        68810     45500   
4         6075020100      CA  San Francisco County        67000     41860   
...              ...     ...                   ...          ...       ...   
1139454  18097360301      IN         Marion County        36003     26900   
1139455  18081610802      IN        Johnson County        25450     26900   
1139456  18097390403      IN         Marion County        36003     26900   
1139457  18097330108      IN         Marion County        36003     26900   
1139458  18097361400      IN         Marion County        36003     26900   

         metdiv10_y  ccflag10_y     MHMVAL70  
0             99999           0  12917.88036  
1             99999           0  7119.412573  
2             99999           0  7736.080911  
3             99999           1  10877.32895  
4             41884           1        21500  
...             ...         ...          ...  
1139454       99999           1  14265.18386  
1139455       99999           0  18207.81447  
1139456       99999           1  19791.66705  
1139457       99999           1  32817.59032  
1139458       99999           1  14776.57597  

[1139459 rows x 17 columns]

In [22]:
ltdb_1970_1980_tract["TRTID10_x"].equals(ltdb_1970_1980_tract["TRTID10_y"])

False

This gets a dataframe with 52759 rows, which is the same as the 1970 data. However, the `TRTID10_x` and `TRTID10_y` variables do not equal each other. So when I join on tracts, some 1970 values are not of the same 2010 tract as the 1980 values. This implies I should use `ltdb_1970_1980` where all tract values do match, and then throw out the entries where there is a 1980 tract but not a 1970 tract. 

# Checking For Empty Values

In [23]:
val_70 = ltdb_1970_1980["MHMVAL70"]
val_70.isna().sum()

0

In [24]:
val_80 = ltdb_1970_1980["MHMVAL80"]
val_80.isna().sum()

2763

So there are 2763 tracts with values in 1980 but not in 1970. Removing these seems the best approach, as this is a relatively small proportion of the overall population.

My final dataset will just include `TRTID10`, `state_x`, `county_x`, `tract_x`, `MHMVAL70`, and `MHMVAL80`. 

In [25]:
ltdb_value = ltdb_1970_1980[["TRTID10", "state_x", "county_x", "tract_x", "MHMVAL70", "MHMVAL80"]]
ltdb_value.isna().sum()

TRTID10        0
state_x        0
county_x       0
tract_x        0
MHMVAL70       0
MHMVAL80    2763
dtype: int64

In [26]:
ltdb_value = ltdb_value.dropna()
ltdb_value.isna().sum()

TRTID10     0
state_x     0
county_x    0
tract_x     0
MHMVAL70    0
MHMVAL80    0
dtype: int64

In [27]:
ltdb_value.to_csv("ltdb_value.csv")

# Joining 1980 and 1990 which I was actually meaning to do in the first place but kinda screwed up

In [39]:
ltdb_1980_1990 = ltdb_1990.merge(ltdb_1980, on="TRTID10")
ltdb_1980_1990

TRTID10 state_x              county_x              tract_x  \
0       1001020100      AL        Autauga County     Census Tract 201   
1       6083002402      CA  Santa Barbara County   Census Tract 24.02   
2       1001020200      AL        Autauga County     Census Tract 202   
3       6083002102      CA  Santa Barbara County   Census Tract 21.02   
4       1001020300      AL        Autauga County     Census Tract 203   
...            ...     ...                   ...                  ...   
59182  26081011804      MI           Kent County  Census Tract 118.04   
59183  26077002901      MI      Kalamazoo County   Census Tract 29.01   
59184  26077002903      MI      Kalamazoo County   Census Tract 29.03   
59185  47149040810      TN     Rutherford County  Census Tract 408.10   
59186  53073010100      WA        Whatcom County     Census Tract 101   

       placefp10_x  cbsa10_x  metdiv10_x  ccflag10_x      MHMVAL90 state_y  \
0            62328     33860       99999           0   62200.00055      AL   
1            69196     42060       99999           1  144999.99920      CA   
2            62328     33860       99999           0   43600.00056      AL   
3            69196     42060       99999           1  157600.00070      CA   
4            62328     33860       99999           0   55200.00044      AL   
...            ...       ...         ...         ...           ...     ...   
59182        99999     24340       99999           0  102400.00330      MI   
59183        99999     28020       99999           0   64899.78331      MI   
59184        99999     28020       99999           0   72088.25791      MI   
59185        99999     34980       99999           0   60999.99927      TN   
59186        53800     13380       99999           0   71300.08319      WA   

                   county_y              tract_y  placefp10_y  cbsa10_y  \
0            Autauga County     Census Tract 201        62328     33860   
1      Santa Barbara County   Census Tract 24.02        69196     42060   
2            Autauga County     Census Tract 202        62328     33860   
3      Santa Barbara County   Census Tract 21.02        69196     42060   
4            Autauga County     Census Tract 203        62328     33860   
...                     ...                  ...          ...       ...   
59182           Kent County  Census Tract 118.04        99999     24340   
59183      Kalamazoo County   Census Tract 29.01        99999     28020   
59184      Kalamazoo County   Census Tract 29.03        99999     28020   
59185     Rutherford County  Census Tract 408.10        99999     34980   
59186        Whatcom County     Census Tract 101        53800     13380   

       metdiv10_y  ccflag10_y     MHMVAL80  
0           99999           0          NaN  
1           99999           1  65120.99963  
2           99999           0  26837.00034  
3           99999           1  66500.00045  
4           99999           0          NaN  
...           ...         ...          ...  
59182       99999           0  58685.99943  
59183       99999           0  45606.85763  
59184       99999           0  58217.46546  
59185       99999           0  40642.00092  
59186       99999           0  46142.94263  

[59187 rows x 17 columns]

In [51]:
ltdb_value_8090 = ltdb_1980_1990[["TRTID10", "state_x", "county_x", "tract_x", "MHMVAL80", "MHMVAL90"]]
ltdb_value_8090.isna().sum()

TRTID10        0
state_x        0
county_x       0
tract_x        0
MHMVAL80    3181
MHMVAL90     218
dtype: int64

In [52]:
val_80 = ltdb_1980_1990["MHMVAL80"]
val_80.isna().sum()

3181

In [53]:
val_90 = ltdb_1980_1990["MHMVAL90"]
val_90.isna().sum()

218

In [55]:
ltdb_value_8090

TRTID10 state_x              county_x              tract_x  \
0       1001020100      AL        Autauga County     Census Tract 201   
1       6083002402      CA  Santa Barbara County   Census Tract 24.02   
2       1001020200      AL        Autauga County     Census Tract 202   
3       6083002102      CA  Santa Barbara County   Census Tract 21.02   
4       1001020300      AL        Autauga County     Census Tract 203   
...            ...     ...                   ...                  ...   
59182  26081011804      MI           Kent County  Census Tract 118.04   
59183  26077002901      MI      Kalamazoo County   Census Tract 29.01   
59184  26077002903      MI      Kalamazoo County   Census Tract 29.03   
59185  47149040810      TN     Rutherford County  Census Tract 408.10   
59186  53073010100      WA        Whatcom County     Census Tract 101   

          MHMVAL80      MHMVAL90  
0              NaN   62200.00055  
1      65120.99963  144999.99920  
2      26837.00034   43600.00056  
3      66500.00045  157600.00070  
4              NaN   55200.00044  
...            ...           ...  
59182  58685.99943  102400.00330  
59183  45606.85763   64899.78331  
59184  58217.46546   72088.25791  
59185  40642.00092   60999.99927  
59186  46142.94263   71300.08319  

[59187 rows x 6 columns]

In [56]:
ltdb_value_8090 = ltdb_value_8090.dropna()
ltdb_value_8090

TRTID10 state_x              county_x              tract_x  \
1       6083002402      CA  Santa Barbara County   Census Tract 24.02   
2       1001020200      AL        Autauga County     Census Tract 202   
3       6083002102      CA  Santa Barbara County   Census Tract 21.02   
5       6083002009      CA  Santa Barbara County   Census Tract 20.09   
6       1001020400      AL        Autauga County     Census Tract 204   
...            ...     ...                   ...                  ...   
59182  26081011804      MI           Kent County  Census Tract 118.04   
59183  26077002901      MI      Kalamazoo County   Census Tract 29.01   
59184  26077002903      MI      Kalamazoo County   Census Tract 29.03   
59185  47149040810      TN     Rutherford County  Census Tract 408.10   
59186  53073010100      WA        Whatcom County     Census Tract 101   

          MHMVAL80      MHMVAL90  
1      65120.99963  144999.99920  
2      26837.00034   43600.00056  
3      66500.00045  157600.00070  
5      78422.00169  184097.26050  
6      43843.99975   67299.99962  
...            ...           ...  
59182  58685.99943  102400.00330  
59183  45606.85763   64899.78331  
59184  58217.46546   72088.25791  
59185  40642.00092   60999.99927  
59186  46142.94263   71300.08319  

[55991 rows x 6 columns]

In [57]:
ltdb_value_8090.to_csv("ltdb_value_8090.csv")

In [45]:
#ltdb_1990_1980 = ltdb_1980.merge(ltdb_1990, on="TRTID10")
#ltdb_1990_1980

In [46]:
#val_80 = ltdb_1990_1980["MHMVAL80"]
#val_80.isna().sum()